Cython/Compiler/ExprNodes.py

```diff
@@ -2808,9 +2808,13 @@ def generate_result_code(self, code):
             # PyObject_GetIter() fails if "tp_iternext" is not set, but the check below
             # makes it visible to the C compiler that the pointer really isn't NULL, so that
             # it can distinguish between the special cases and the generic case
+            code.putln("#if CYTHON_USE_TYPE_SLOTS")
             code.putln("%s = Py_TYPE(%s)->tp_iternext; %s" % (
```

Comment:
Yeah, we should guard this as well. There are more places like this, though. I've push cd3ce37dfbff134baa110437a0597cd5f8663d97 to master.